src/librustc/hir/map/mod.rs

```diff
@@ -1352,10 +1353,8 @@ fn hir_id_to_string(map: &Map<'_>, id: HirId, include_id: bool) -> String {
 
 pub fn provide(providers: &mut Providers<'_>) {
     providers.def_kind = |tcx, def_id| {
-        if let Some(hir_id) = tcx.hir().as_local_hir_id(def_id) {
-            tcx.hir().def_kind(hir_id)
-        } else {
-            bug!("calling local def_kind query provider for upstream DefId: {:?}", def_id);
```

Comment:
This is kind of a silly (pre-existing) implementation, you should be able to just use `def_id.to_local()` and letting it panic instead of this whole thing.

Reply:
AFAICT `to_local` doesn't panic. It doesn't sanity check anything. That's why I made `try_to_local`.

Reply:
`to_panic` calls `LocalDefId::from_def_id` which asserts. This will be better post-#66131, which I've been working on for the past day now.